### PR TITLE
Update the RHMI Operator to make use of the secret being created by t…

### PR DESCRIPTION
**Pull request by Rajagopalan-Ranganathan**
# Description
Update the RHMI Operator to make use of the secret being created by the DeadMansSnitch Operator

JIRA Link: https://issues.redhat.com/browse/INTLY-9096

Code changes made to use the secret created by the DeadMansSnitch operator first, and if not
then use the secret created by the cssre.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/products/monitoring/reconciler.go [hunk 1]
```diff
@@ -644,13 +644,10 @@ func (r *Reconciler) reconcileAlertManagerConfigSecret(ctx context.Context, serv
...
 	}
 
-	// handle dead mans snitch credentials
-	dmsSecret := &corev1.Secret{}
-	if err := serverClient.Get(ctx, types.NamespacedName{Name: r.installation.Spec.DeadMansSnitchSecret, Namespace: rhmiOperatorNs}, dmsSecret); err != nil {
-		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("could not obtain dead mans snitch credentials secret: %w", err)
-	}
-	if len(dmsSecret.Data["url"]) == 0 {
-		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("url is undefined in dead mans switch secret")
+	//Get dms  credentials
```
Comment (obrienrobert) [inline pkg/products/monitoring/reconciler.go:647]: Tiny little detail -> 2 white spaces between `dms` and `credentials` in the comment.

File: pkg/products/monitoring/reconciler.go [hunk 2]
```diff
@@ -776,3 +773,28 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8s
...
+
+	if err != nil {
+		return "", fmt.Errorf("could not obtain dead mans snitch credentials secret: %w", err)
+	}
+
+	if len(dmsSecret.Data["SNITCH_URL"]) != 0 {
+		secret = string(dmsSecret.Data["SNITCH_URL"])
+	} else if len(dmsSecret.Data["url"]) != 0 {
+		secret = string(dmsSecret.Data["url"])
+	}
```
Comment (obrienrobert) [inline pkg/products/monitoring/reconciler.go:793]: Would it work if we placed an `else` statement here and returned if the secret hasn't been assigned? i.e:

 ```
  else {
   return secret, fmt.Errorf("url is undefined in dead mans snitch secret")
  }
```

File: pkg/products/monitoring/reconciler.go (hunk 1 above)
Comment (Rajagopalan-Ranganathan) [reply pkg/products/monitoring/reconciler.go:647]: Agreed. Fixed.

File: pkg/products/monitoring/reconciler.go (hunk 2 above)
Comment (Rajagopalan-Ranganathan) [reply pkg/products/monitoring/reconciler.go:793]: It would work. Agreed and moved the code.

**Comment by david-martin** [comment]
/approve

@obrienrobert can you try this out and give lgtm?

**Comment by obrienrobert** [comment]
Verified using the manual test steps in `m05-verify-secrets.md`.

/lgtm

**Pull request merged**